sycl/test/check_device_code/vector/vector_math_ops.cpp

```diff
@@ -104,6 +165,27 @@ SYCL_EXTERNAL auto TestAdd(vec<bool, 4> a, vec<bool, 4> b) { return a + b; }
 // CHECK-NEXT:    [[EXTRACTVEC5_I:%.*]] = shufflevector <4 x half> [[TMP0]], <4 x half> poison, <4 x i32> <i32 0, i32 1, i32 2, i32 poison>
 // CHECK-NEXT:    store <4 x half> [[EXTRACTVEC5_I]], ptr addrspace(4) [[AGG_RESULT]], align 8, {{.*}}
 // CHECK-NEXT:    ret void
+// CHECK-ARR-STORAGE-LABEL: define dso_local spir_func void @_Z7TestAddN4sycl3_V13vecINS0_6detail9half_impl4halfELi3EEES5_(
+// CHECK-ARR-STORAGE-SAME: ptr addrspace(4) dead_on_unwind noalias nocapture writable writeonly sret(%"class.sycl::_V1::vec.8") align 8 [[AGG_RESULT:%.*]], ptr nocapture noundef readonly byval(%"class.sycl::_V1::vec.8") align 8 [[A:%.*]], ptr nocapture noundef readonly byval(%"class.sycl::_V1::vec.8") align 8 [[B:%.*]]) {{.*}} {
+// CHECK-ARR-STORAGE-NEXT:  entry:
+// CHECK-ARR-STORAGE-NEXT:    tail call void @llvm.experimental.noalias.{{.*}}
+// CHECK-ARR-STORAGE-NEXT:    br label [[FOR_COND_I:%.*]]
+// CHECK-ARR-STORAGE:       for.cond.i:
+// CHECK-ARR-STORAGE-NEXT:    [[I_0_I:%.*]] = phi i64 [ 0, [[ENTRY:%.*]] ], [ [[INC_I:%.*]], [[FOR_BODY_I:%.*]] ]
```

Comment:
Why do we have a for-loop for half?
For half, can we bit cast std::array<sycl::half, 2> to _Float16 ext_vector_type(2) ? Is the bit pattern for _Float16 same as sycl::half? Perhaps, because sycl::half uses _Float16 underneath.
But, `[[LOADVEC4_I:%.*]] = load <4 x half>, ptr [[A]], align 8, {{.*}}`  loads a vector of **LLVM type half** from vector of  _Float16? Smells like a bug...